### PR TITLE
Transaction detection improvements

### PR DESCRIPTION
**Pull request by Nikita-Polyakov**
https://soramitsu.atlassian.net/browse/PW-309
https://soramitsu.atlassian.net/browse/PW-587

### CODE REVIEW

File: src/components/WalletHistory.vue [hunk 1]
```diff
@@ -189,9 +189,9 @@ export default class WalletHistory extends Mixins(LoadingMixin, TransactionMixin
   }
 
   getStatus(status: string): string {
-    if ([TransactionStatus.Error, 'invalid'].includes(status)) {
+    if ([TransactionStatus.Error, TransactionStatus.Invalid].includes(status as TransactionStatus)) {
       status = TransactionStatus.Error;
-    } else if (status !== TransactionStatus.Finalized) {
+    } else if (![TransactionStatus.InBlock, TransactionStatus.Finalized].includes(status as TransactionStatus)) {
```
Comment (stefashkaa) [inline src/components/WalletHistory.vue:194]: !isFinalizedStatus(status)

File: src/components/WalletTransactionDetails.vue [hunk 2]
```diff
@@ -115,11 +115,17 @@ export default class WalletTransactionDetails extends Mixins(TranslationMixin, N
   }
 
   get isComplete(): boolean {
-    return [TransactionStatus.Finalized, 'done'].includes(this.selectedTransaction.status as TransactionStatus);
+    return [TransactionStatus.InBlock, TransactionStatus.Finalized, 'done'].includes(
```
Comment (stefashkaa) [inline src/components/WalletTransactionDetails.vue:118]: `done`?? I forgot, why did we put it here

File: src/components/mixins/TransactionMixin.ts [hunk 3]
```diff
@@ -84,60 +81,61 @@ export default class TransactionMixin extends Mixins(TranslationMixin, LoadingMi
...
-      ![TransactionStatus.Finalized, TransactionStatus.Error].includes(value.status as TransactionStatus)
+      ![TransactionStatus.InBlock, TransactionStatus.Finalized, TransactionStatus.Error].includes(
+        value.status as TransactionStatus
+      )
     ) {
       return;
     }
+
     const message = this.getMessage(value);
+    const isNextTx = !oldValue || oldValue.id !== value.id;
```
Comment (stefashkaa) [inline src/components/mixins/TransactionMixin.ts:106]: please add a comment why do we need this flag

**Review by stefashkaa — APPROVED** [review]

File: src/components/WalletTransactionDetails.vue (hunk 2 above)
Comment (Nikita-Polyakov) [reply src/components/WalletTransactionDetails.vue]: I didn't found this status in available transaction statuses (checked it in codec), so i removed it from here

**Pull request merged**